code/modules/wod13/onyxcombat.dm

```diff
@@ -578,3 +551,83 @@
 				elysium_checks = 0
 				if(HAS_TRAIT(src, TRAIT_ELYSIUM))
 					REMOVE_TRAIT(src, TRAIT_ELYSIUM, "elysium")
+
+/mob/living/carbon/human
+	var/current_police_raid = FALSE //lock to prevent multiple handle_police_raid() procs from going at the same time
+
+/mob/living/carbon/human/proc/handle_police_raid() //hooo boy
+	if(warrant)
+		if(key) //are we a player?
+			if(stat != DEAD) //are we Not Dead?
+				if(istype(get_area(src), /area/vtm))
+					var/area/vtm/V = get_area(src)
+					if(V.upper && !current_police_raid && last_raid+next_raid < world.time)
+						current_police_raid = TRUE
+						var/warning_msg = pick("You see them searching above...", "The rumble of helicopter blades draws closer...")
+						to_chat(src, "<span class='warning'>[warning_msg]</span>")
+						playsound(loc, 'code/modules/wod13/sounds/helicopter_far.ogg', 50, FALSE)
+						sleep(60)
```

Comment:
Could you use addtimer instead of Sleep(x)? If I'm reading the code right, this will cause the entire life() proc to sleep on this function, when it should be doing an asynchronous call 60 seconds later, then a second one 40 seconds after that.